server/src/main/java/org/opensearch/search/aggregations/bucket/composite/UnsignedLongValuesSource.java

```diff
@@ -14,21 +14,35 @@
 import org.apache.lucene.search.PointRangeQuery;
 import org.apache.lucene.search.Query;
 import org.opensearch.common.CheckedFunction;
+import org.opensearch.common.Numbers;
+import org.opensearch.common.lease.Releasables;
 import org.opensearch.common.util.BigArrays;
+import org.opensearch.common.util.BitArray;
+import org.opensearch.common.util.LongArray;
 import org.opensearch.index.mapper.MappedFieldType;
 import org.opensearch.index.mapper.NumberFieldMapper;
 import org.opensearch.search.DocValueFormat;
+import org.opensearch.search.aggregations.LeafBucketCollector;
 import org.opensearch.search.aggregations.bucket.missing.MissingOrder;
 
 import java.io.IOException;
-import java.util.function.LongUnaryOperator;
+import java.math.BigInteger;
+import java.util.Objects;
 
 /**
  * A {@link SingleDimensionValuesSource} for unsigned longs.
  *
  * @opensearch.internal
  */
-public class UnsignedLongValuesSource extends LongValuesSource {
```

Comment:
Subclassing `LongValuesSource` turned out to be not a good idea, primarily because `toComparable` that returned `Long` (but should `BigInteger` actually), messing up the ordering sometimes.